Toolbar: unify Storybook examples under one file, migrate from knobs to controls

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Following up the prep work from #46951, this PR tidies up `Toolbar`'s Storybook examples into one file, and migrates from knobs to controls (even if no controls are being displayed at this point)

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

The knobs addon is currently blocking using `npm 8` (see #46443)

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

- Removed knobs in favor of controls
- Move each example to its separate story
- Delete redundant stories

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

Open Storybook, check that stories behave as expected and are comprehensive.

ciampo committed:
Delete unnecessary storibook examples

ciampo committed:
Rewrite settings (knobs to controls, add subcomponents)

ciampo committed:
Split each example to its separate story

mirka commented on packages/components/src/toolbar/stories/index.js:
```diff
@@ -98,72 +122,99 @@ export const _default = () => {
-					<ToolbarButton icon={ link } title="Link" />
-				</Toolbar>
-			</div>
+		// id is required for server side rendering
+		// eslint-disable-next-line no-restricted-syntax
+		<Toolbar>
```
Copypasta?

ciampo replied:
Yup! Not an issue anymore since I deleted all of those extra stories

mirka commented on packages/components/src/toolbar/stories/index.js:
```diff
@@ -30,10 +54,10 @@ function InlineImageIcon() {
 
-/* eslint-disable no-restricted-syntax */
-export const _default = () => {
+export const Default = () => {
```
This isn't super important because `Toolbar` itself doesn't have a lot of props anyway, but I think a general pattern we can use to keep the main component "controls-ready" is to pass in all the content via the `children` arg.

```jsx
const Template = ( props ) => <Toolbar { ...props } />;

export const Foo = Template.bind( {} );
Foo.args = {
	label: 'Options',
	children: <ToolbarGroup>{/* whatever */}</ToolbarGroup>,
};

ciampo replied:
Good point, I refactored the stories to use `Template.bind()` + `args` syntax as suggested

mirka commented on packages/components/src/toolbar/stories/index.js:
```diff
@@ -12,15 +12,39 @@ import {
+	ToolbarItem,
+	ToolbarDropdownMenu,
+} from '..';
+import { SVG, Path, DropdownMenu } from '../../../build';
```
Wrong import path?

ciampo replied:
Interesting glitch of my auto-import path plugin. It's fixed now

mirka reviewed (commented):
Personally I don't see too much value in most of the stories except Without Group, because all the use cases seem to be covered in the Default story. I'm hoping we can think of a better documentation strategy once we have TS types for the subcomponents!

Looks good to merge once the import path and copypasta is cleaned up 👍

ciampo committed:
Remove redundant stories

ciampo committed:
Add text+icon example to default story

ciampo committed:
Fix import path

ciampo committed:
Use template and args syntax

ciampo commented:
> Personally I don't see too much value in most of the stories except Without Group, because all the use cases seem to be covered in the Default story.

Good point, I removed those extra stories. I just added an example of `ToolbarButton` with both icon _and_ text to the default example, the rest (as you mentioned) was already covered.

Can you give this PR a final look? 🙏

mirka reviewed (approved):
I just realized the default story is also suffering from the [dropdown cutoff problem](https://github.com/WordPress/gutenberg/pull/47149#discussion_r1071102197) 😕 I wish there was a better automatic fix but I haven't found one unfortunately. Last time I checked, we'd need to switch to fully custom Docs pages to get the popover portal where we want it.

Otherwise this looks good to go 🚢

ciampo committed:
Avoid live demo being cut off in docs view
